### PR TITLE
Add Databricks Vector Search example notebook and documentation

### PR DESCRIPTION
**Pull request by freemso**
This PR adds an example notebook for the Databricks Vector Search vector store. It also adds an introduction to the Databricks Vector Search product on the Databricks's provider page.

### CODE REVIEW

File: docs/docs/integrations/vectorstores/databricks_vector_search.ipynb [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+    "from databricks.vector_search.client import VectorSearchClient\n",
+    "\n",
+    "vsc = VectorSearchClient()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create a Vector Search Endpoint"
```
Comment (ninabacc-db) [inline docs/docs/integrations/vectorstores/databricks_vector_search.ipynb:169]: Add a sentence here that states what this endpoint is for.

File: docs/docs/integrations/vectorstores/databricks_vector_search.ipynb [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {},
+     "inputWidgets": {},
+     "nuid": "81090f87-3efd-4c1e-9f58-8d6adba7553d",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "## Create Direct Vector Access Index"
```
Comment (ninabacc-db) [inline docs/docs/integrations/vectorstores/databricks_vector_search.ipynb:193]: Add content here describing what a Direct Vector Access Index is. We want to make sure customers know the difference versus what a Delta Sync Index does, especially what customers are responsible for bringing for the Direct Vector Access Index to work

File: docs/docs/integrations/vectorstores/databricks_vector_search.ipynb [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+     "inputWidgets": {},
+     "nuid": "46e3f41b-dac2-4bed-91cb-a3914c25d275",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "## Work with Delta Sync Index\n",
+    "\n",
+    "You can also use `DatabricksVectorSearch` to search in a Delta Sync Index. Delta Sync Index automatically syncs from a Delta table. You don't need to call `add_text`/`add_documents` manually."
```
Comment (ninabacc-db) [inline docs/docs/integrations/vectorstores/databricks_vector_search.ipynb:331]: Maybe we add a link to the Databricks docs here for the definition of what a Delta Sync Index is and what is does on behalf of customers. Feel like we are glossing over what customers need to set up on their end if they want to use this index type

**Review by ninabacc-db — COMMENTED** [review]
Couple comments, after those are addressed I can approve

File: docs/docs/integrations/vectorstores/databricks_vector_search.ipynb (hunk 1 above)
Comment (freemso) [reply docs/docs/integrations/vectorstores/databricks_vector_search.ipynb]: done!

File: docs/docs/integrations/vectorstores/databricks_vector_search.ipynb (hunk 2 above)
Comment (freemso) [reply docs/docs/integrations/vectorstores/databricks_vector_search.ipynb]: done!

File: docs/docs/integrations/vectorstores/databricks_vector_search.ipynb (hunk 3 above)
Comment (freemso) [reply docs/docs/integrations/vectorstores/databricks_vector_search.ipynb]: Added a link.

**Review by ninabacc-db — APPROVED** [review]
LGTM

**Comment by freemso** [comment]
@baskaryan The PR is ready for review. Thanks!

**Pull request merged**